run/cli_ioproducer.go

```diff
@@ -0,0 +1,35 @@
+package run
+
+import (
+	"context"
+	"io"
+	"log"
+	"os"
+)
+
+// CliIOProducer is the IOProducer for the CliRunner. The input and output paths
+// are used to configure the input and output readers and writers. If the paths
+// are empty, os.Stdin and os.Stdout are used.
+func CliIOProducer(_ context.Context, cfg CLIRunnerConfig) IOData {
+	reader := os.Stdin
+	if cfg.Runner.Input.Path != "" {
+		r, err := os.Open(cfg.Runner.Input.Path)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
I wonder if `log.Fatal` here is the right thing to do. That means the user of that function cannot recover from a panic and no `defer` calls are executed. Applies to other `log.Fatal` calls as well.

Reply:
I introduced an `error` on the `IOProducer` so that users can handle that error in their user code.